src/otx/algo/samplers/repeat_sampler.py

```diff
@@ -0,0 +1,137 @@
+"""OTX sampler."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+import torch
+from torch.utils.data.sampler import Sampler
+
+if TYPE_CHECKING:
+    from otx.core.data.dataset.base import OTXDataset
+
+
+def get_proper_repeat_times(
+    data_size: int,
+    batch_size: int,
+    coef: float,
+    min_repeat: float,
+) -> float:
+    """Get proper repeat times for adaptive training.
+
+    Args:
+        data_size (int): The total number of the training dataset
+        batch_size (int): The batch size for the training data loader
+        coef (float) : coefficient that effects to number of repeats
+                       (coef * math.sqrt(num_iters-1)) +5
+        min_repeat (float) : minimum repeats
+    """
+    if data_size == 0 or batch_size == 0:
+        return 1
+    n_iters_per_epoch = math.ceil(data_size / batch_size)
+    return math.floor(max(coef * math.sqrt(n_iters_per_epoch - 1) + 5, min_repeat))
+
+
+class RepeatSampler(Sampler):
```

Comment:
Why do we need this although we can set the sample size of `torch.utils.data.RandomSampler` with `num_samples` argument?

Reply:
@sungmanc I'm not exactly sure of the history of the use of that class, and I think Vinnam's comment is right on the money. From my quick look, it doesn't seem to be much different than `RandomSampler` with `num_samples`, other than supporting the code snippet below, but do you have any thoughts?
![image](https://github.com/openvinotoolkit/training_extensions/assets/38045080/806f441b-546c-4f3e-b618-e75d492eefd2)

Reply:
I just remembered the reason why we brought this class: we want to add the functionality that adjusts the repeat with float times. i.e. (1.2x num_data) or (0.7x num_data). However, since we don't have a room, this is the same with the `torch.RandomSampler`.  So, for the current status, I think using the `torch.RandomSampler` looks good.